File: 0946-validate-stack-sequences/0946-validate-stack-sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        
        stack<int>s;
        int j = 0;
        for(int i = 0; i<pushed.size(); i++){
            s.push(pushed[i]);
            while(!s.empty() && s.top() == popped[j] && j < popped.size()){
                s.pop();
                j++;
            }
        }
        
        if(!s.empty() || j != popped.size())
            return false;
        return true;
    }
};

File: 0946-validate-stack-sequences/0946-validate-stack-sequences.js
/**
 * @param {number[]} pushed
 * @param {number[]} popped
 * @return {boolean}
 */
var validateStackSequences = function(pushed, popped) {
    
    let stack = [];
    let j = 0;
    pushed.forEach(e=>{
        stack.push(e);
        while(stack.length>0 && stack.slice(-1) == popped[j]){
            stack.pop();
            j++;
        }
    })
    if(stack.length>0 || j != popped.length) return false;
    return true;
};